pytest_html/plugin.py

```diff
@@ -83,97 +84,117 @@ def __init__(self, logfile):
         logfile = os.path.expanduser(os.path.expandvars(logfile))
         self.logfile = os.path.abspath(logfile)
         self.test_logs = []
+        self.tests = []
         self.errors = self.failed = 0
         self.passed = self.skipped = 0
         self.xfailed = self.xpassed = 0
         self.rerun = 0
 
-    def append_extra_html(self, extra, additional_html, links_html):
-        href = None
-        if extra.get('format') == extras.FORMAT_IMAGE:
-            href = '#'
-            image = 'data:image/png;base64,{0}'.format(
-                    extra.get('content'))
-            additional_html.append(html.div(
-                html.a(html.img(src=image), href="#"),
-                class_='image'))
-
-        elif extra.get('format') == extras.FORMAT_HTML:
-            additional_html.append(html.div(raw(extra.get('content'))))
-
-        elif extra.get('format') == extras.FORMAT_JSON:
-            href = data_uri(json.dumps(extra.get('content')),
-                            mime_type='application/json')
-
-        elif extra.get('format') == extras.FORMAT_TEXT:
-            href = data_uri(extra.get('content'))
-
-        elif extra.get('format') == extras.FORMAT_URL:
-            href = extra.get('content')
-
-        if href is not None:
-            links_html.append(html.a(
-                extra.get('name'),
-                class_=extra.get('format'),
-                href=href,
-                target='_blank'))
-            links_html.append(' ')
-
-    def append_log_html(self, report, additional_html):
-        log = html.div(class_='log')
-        if report.longrepr:
-            for line in str(report.longrepr).splitlines():
-                if not PY3:
-                    line = line.decode('utf-8')
-                separator = line.startswith('_ ' * 10)
-                if separator:
-                    log.append(line[:80])
-                else:
-                    exception = line.startswith("E   ")
-                    if exception:
-                        log.append(html.span(raw(escape(line)),
-                                             class_='error'))
+    class TestReport:
```

Comment:
I'd prefer if we named this `TestResult` as it represents a single result rather than a full report.